Fix ClusterOperator tests that can only report failures

dgoodwin opened the pull request:
We have discovered that if there are no clusteroperator intervals in the
e2e/conformance test portion, this test is omitted entirely which breaks
test pass rates in a variety of tooling, as well as the aggregator. In
this case if the operator does trigger an unexpected transition in the
conformance stage, a fail is recorded, and the other jobs in aggregation
report no result, resulting in < 6 results to aggregate on and a
failure.

Switch to always checking for, and including tests for, all operators,
regardless if they have recorded any intervals.

[TRT-1100](https://issues.redhat.com//browse/TRT-1100)

stbenjam commented on pkg/synthetictests/operators.go:
```diff
@@ -39,7 +39,7 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
 	duration := stop.Sub(start).Seconds()
 
-	knownOperators := allOperators(events)
+	knownOperators := sets.NewString(platformidentification.KnownOperators.List()...)
```
allOperators was already using this list, but then adding any new operators seen in events.

This change will have the effect of killing off the control-plane-machine-set test entirely, because it's not in https://github.com/openshift/origin/blob/master/pkg/synthetictests/platformidentification/operator_mapping.go#L87-L120

Can we update that file too? I also think you can get rid of the intermediate variable and just reference `platformidentification.KnownOperators.List()` on line 46 here

dgoodwin commented:
Added control-plane-machine-set to known operators, removed the interim variable, but kept the change to no longer add in other operators we see, as this would lead to tests that are not always present with wonky pass rates.

stbenjam commented:
/lgtm
